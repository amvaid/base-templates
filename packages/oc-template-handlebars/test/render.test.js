const render = require('../lib/render');

describe('render method', () => {
  describe('when invoked with a valid template', () => {
    const model = { aModel: true };
    const template = {
      compiler: [7, '>= 4.0.0'],
      main: function() {
        return 'Hello world!';
      },
      useData: !0
    };
    const callback = jest.fn();

    render({ model, template }, callback);
    test('should correctly invoke the callback', () => {
      expect(callback).toBeCalledWith(null, 'Hello world!');
    });
  });

  describe('when invoked with a broken view-model that throws an exception', () => {
    const model = { aModel: true };
    const template = {
      compiler: [7, '>= 4.0.0'],
      main: function() {
        throw new Error('blargh');
      },
      useData: !0
    };
    const callback = jest.fn();

    render({ model, template }, callback);
    test('should correctly invoke the callback', () => {
      expect(callback).toBeCalledWith(new Error('blargh'));
    });
  });
});
