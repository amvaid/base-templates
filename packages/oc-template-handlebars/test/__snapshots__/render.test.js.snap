// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render method when invoked with a valid template with valid view-models should correctly render with viewmodel = {name:"Jane"} 1`] = `
"<img src=\\"img/logo.png\\" />

<div>Hello Jane</div>
"
`;

exports[`render method when invoked with a valid template with valid view-models should correctly render with viewmodel = {name:"John"} 1`] = `
"<img src=\\"img/logo.png\\" />

<div>Hello John</div>
"
`;

exports[`render method when invoked with a valid template with valid view-models should correctly render with viewmodel = {name:"Mike"} 1`] = `
"<img src=\\"img/logo.png\\" />

<div>Hello Mike</div>
"
`;
